File: java/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/polygonTriangulation/monotoneDecomposition/_ContourData.java
package vsdk.toolkit.environment.geometry.geometricProcessing.polygonTriangulation.monotoneDecomposition;

import java.util.ArrayList;

final class _ContourData {
    final ArrayList<_IndexedVertex> vertices = new ArrayList<>();
    final ArrayList<Integer> childContours = new ArrayList<>();
    double signedArea;
    int parentContour = -1;
    int depth = -1;
}

File: typescript/base/src/main/vsdk/toolkit/environment/geometry/geometricProcessing/polygonTriangulation/monotoneDecomposition/_ContourData.ts
import { _IndexedVertex } from "./_IndexedVertex.js";
export class _ContourData {
    public readonly vertices: _IndexedVertex[] = [];
    public readonly childContours: number[] = [];
    public signedArea = 0;
    public parentContour = -1;
    public depth = -1;
}
